coreneuron/mpi/nrnmpi.cpp

```diff
@@ -73,7 +74,7 @@ void nrnmpi_init(int nrnmpi_under_nrncontrol, int* pargc, char*** pargv) {
                 b = true;
                 break;
             }
-            if (strcmp("--mpi", (*pargv)[i]) == 0) {
+            if (corenrn_param.mpi_enable) {
```

Comment:
Can the enclosing loop be modified to smtg like: 

```cpp 
if (corenrn_param.mpi_enable || corenrn_param.p4)
{
 b = true;
}
```

Reply:
The problem with **p4** is that it does not exists anymore (coreneuron side at least), `-p` (for `--pattern`) take a file path.

Reply:
At the very least `corenrn_param.mpi_enable` can be checked outside the loop

Reply:
ahahaha of course, I'm so disappointed by this piece of code, that I miss that xD